AutoCollection/Statsbeat.ts

```diff
@@ -148,15 +164,52 @@ class Statsbeat {
         this._retryCount++;
     }
 
-    public async trackStatsbeatMetrics() {
+    public async trackShortIntervalStatsbeats() {
+        this._getResourceProvider(async () => {
+            let networkProperties = {
+                "os": this._os,
+                "rp": this._resourceProvider,
+                "cikey": this._cikey,
+                "runtimeVersion": this._runtimeVersion,
+                "language": this._language,
+                "version": this._sdkVersion,
+                "attach": this._attach,
+                "instrumentation": this._instrumentations,
+            }
+            this._trackRequestDuration(networkProperties);
+            this._trackRequestsCount(networkProperties);
+            let attachProperties = {
+                "os": this._os,
+                "rp": this._resourceProvider,
+                "rpid": this._resourceIdentifier,
```

Comment:
Curious, what will `undefined` serialize as? Empty string? This is for the case when `rp` == `unknown`

Reply:
this._resourceIdentifier defaults to "unknown" so it supposed to never be undefined, are you seeing something here?
I believe undefined properties are dropped when serializing JSON, at least they are when running this in the browser
JSON.stringify({"test":undefined})
"{}"

Reply:
So on non-attach scenario, the payload would look like: {"rp": "unknown"} with no "rpid" correct?

Reply:
Actually rpid is also included, something like this {"rp": "unknown","rpid": "unknown"}